addons/mail/controllers/discuss/channel.py

```diff
@@ -98,9 +98,14 @@ def discuss_channel_messages(self, channel_id, fetch_params=None):
         messages = res.pop("messages")
         if not request.env.user._is_public():
             messages.set_message_done()
+        store = Store(messages, for_current_user=True)
+        for message in messages:
+            if message.is_current_user_or_guest_author:
+                # sudo: mail.message - reading attachments on accessible message is allowed
+                store.add(message.sudo().attachment_ids, "as_author_access_token")
```

Comment:
We should be using store.add on the full recordset with a list of fields rather than making the loop manually.
Attr value can be a compute function, predicate param can be used to filters which fields are returned for each record

Reply:
See https://www.odoo.com/odoo/1519/tasks/4775128 for more context